### PR TITLE
Added some more information about errors on the onRoutingFailure

### PR DESCRIPTION
**Pull request by bilthon**
Currently, the onRoutingFailure callback method does not provide much useful information about the nature of the error. It could be that the API KEY is incorrect, or that the daily quota has been reached, or a bunch of other reasons. The JSON response coming from the Google servers indeed inform the user about the nature of the problem, but the library doesn't relay that information. This commit attempt to fix that.

PS. I noticed the previous pull request had some useless debug logs, thus I'm submitting another updated pull request.

### CODE REVIEW

**Comment by fcduarte** [comment]
@bilthon could you please format the whole code? Using Android Studio's default is fine.

File: library/src/main/java/com/directions/route/AbstractRouting.java [hunk 1]
```diff
@@ -111,7 +114,13 @@ private void dispatchOnCancelled() {
      */
     @Override
     protected ArrayList<Route> doInBackground(Void... voids) {
-        return new GoogleParser(constructURL()).parse();
+        ArrayList<Route> result = null;
```
Comment (fcduarte) [inline library/src/main/java/com/directions/route/AbstractRouting.java]: As a best pratice, create an empty list. `new GoogleParser(constructURL()).parse();` returns an empty list if it doesn't find anything.

File: library/src/main/java/com/directions/route/AbstractRouting.java [hunk 2]
```diff
@@ -123,9 +132,9 @@ protected void onPreExecute() {
 
     @Override
     protected void onPostExecute(ArrayList<Route> result) {
-        if (result == null) {
-            dispatchOnFailure();
-        } else {
+        if(mException != null){
+            dispatchOnFailure(mException);
+        } else if (result != null) {
```
Comment (fcduarte) [inline library/src/main/java/com/directions/route/AbstractRouting.java]: Just check if `!result.isEmpty()`

File: library/src/main/java/com/directions/route/RouteException.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+    private String statusCode;
+    private String message;
+
+    public RouteException(JSONObject json){
+        if(json == null){
+            statusCode = "";
+            message = "Parsing error";
+            return;
+        }else{
```
Comment (fcduarte) [inline library/src/main/java/com/directions/route/RouteException.java]: if you're exiting the method on the previous line you don't need this `else`

File: sample/src/main/java/com/directions/sample/MainActivity.java [hunk 4]
```diff
@@ -362,10 +362,13 @@ public void route()
 
 
     @Override
-    public void onRoutingFailure() {
+    public void onRoutingFailure(RouteException e) {
         // The Routing request failed
         progressDialog.dismiss();
-        Toast.makeText(this,"Something went wrong, Try again", Toast.LENGTH_SHORT).show();
+        if(e != null)
```
Comment (fcduarte) [inline sample/src/main/java/com/directions/sample/MainActivity.java]: please add braces

**Comment by fcduarte** [comment]
A few comments but it looks good to me. Wait for @jd-alexander to jump in the review.

**Comment by fcduarte** [comment]
nice work BTW, if you found any bug related to the API key please file an issue.

**Comment by bilthon** [comment]
@fcduarte I'm not sure I understand what you mean by "format the whole code". I addressed all the points you raised. Thanks for the feedback!

**Comment by jd-alexander** [comment]
I'm gonna get some time to review this on the weekend and then i will merge to master and updates docs and maven :)  great work @bilthon . Thanks for the review @fcduarte :+1:

**Pull request merged**